kratos/utilities/math_utils.h

```diff
@@ -441,32 +465,32 @@ class MathUtils
             InvertMatrix4(InputMatrix, InvertedMatrix, InputMatrixDet);
         }
         else
-        {   
-            const unsigned int size1 = InputMatrix.size1();
-            const unsigned int size2 = InputMatrix.size2();
+        {
+            const SizeType size1 = InputMatrix.size1();
+            const SizeType size2 = InputMatrix.size2();
             if(InvertedMatrix.size1() != size1 || InvertedMatrix.size2() != size2)
             {
                 InvertedMatrix.resize(size1, size2,false);
             }
-            
+
             int singular = 0;
             typedef permutation_matrix<SizeType> pmatrix;
             Matrix A(InputMatrix);
             pmatrix pm(A.size1());
             singular = lu_factorize(A,pm);
             InvertedMatrix.assign( IdentityMatrix(A.size1()));
             lu_substitute(A, pm, InvertedMatrix);
-             
+
             // Calculating determinant
             InputMatrixDet = 1.0;
-            
-            for (unsigned int i = 0; i < A.size1();i++)
+
+            for (IndexType i = 0; i < A.size1();i++)
             {
-                unsigned int ki = pm[i] == i ? 0 : 1;
+                IndexType ki = pm[i] == i ? 0 : 1;
```

Comment:
it is not really part of this PR, however since i see it i post it here

what is done here on the next line? this looks to me VERY inefficient ... the determinant should be directly the product of the diagonal terms, without the pow. are u really sure the pow is needed?

Reply:
I cannot remember, I did that long time ago :P

Reply:
https://en.wikipedia.org/wiki/LU_decomposition#Computing_the_determinant according to this it is jot necessary

Reply:
~~~
    double det = A[0][0];

    for (int i = 1; i < N; i++)
        det *= A[i][i];

    if ((P[N] - N) % 2 == 0)
        return det; 
    else
        return -det;
~~~

Reply:
in the code snyppet you put, the if is there only once... in the one you implemented it is there for every row...

Reply:
Yeah, but the P and N are unkown if I use the lu_substitute

Reply:
That's why

Reply:
In any case I got rid of the std::pow